internal/internal_workflow_client.go

```diff
@@ -901,6 +917,27 @@ type QueryWorkflowWithOptionsResponse struct {
 	QueryRejected *querypb.QueryRejected
 }
 
+type WorkflowExecutionMetadata struct {
```

Comment:
I picked fields to expsoe here based of what dotnet exposed https://github.com/temporalio/sdk-dotnet/blob/main/src/Temporalio/Client/WorkflowExecutionDescription.cs#L10

Reply:
Works for me, though not the biggest fan of the term "metadata", but I know it's what Java uses, so up to you. I figure one day we'll deprecate all the bad forms of "list" and have a new "list" call that returns `iter.Seq[WorkflowExecutionMetadata]`, but not sure if that affects your name choice.

Reply:
If you have a different name I am open to suggestions, when we add list I was going to add another type that would also embedded `WorkflowExecutionMetadata `

Reply:
Yeah, since `WorkflowExecution` is already in use elsewhere, this is fine. Not sure at list time we need another type since list contents should be a subset of describe, but can discuss at that time.